patches/api/0457-Paper-Item-API.patch

```diff
@@ -0,0 +1,582 @@
++
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/ItemProperties.java b/src/main/java/io/papermc/paper/inventory/item/properties/ItemProperties.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..dde98d97f272a0dda5db4c2598cbc6c982b67c47
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/ItemProperties.java
+@@ -0,0 +1,129 @@
++package io.papermc.paper.inventory.item.properties;
++
++import com.destroystokyo.paper.Namespaced;
++import com.destroystokyo.paper.profile.PlayerProfile;
++import io.leangen.geantyref.TypeFactory;
++import io.papermc.paper.math.Position;
++import io.papermc.paper.property.MutablePropertyView;
++import io.papermc.paper.property.PropertyKey;
++import net.kyori.adventure.text.Component;
++import org.bukkit.Color;
++import org.bukkit.DyeColor;
++import org.bukkit.FireworkEffect;
++import org.bukkit.MusicInstrument;
++import org.bukkit.NamespacedKey;
++import org.bukkit.entity.Axolotl;
++import org.bukkit.inventory.ItemFlag;
++import org.bukkit.inventory.ItemStack;
++import org.bukkit.inventory.meta.BookMeta;
++import org.bukkit.inventory.meta.trim.ArmorTrim;
++import org.bukkit.potion.PotionData;
++import org.bukkit.potion.PotionEffect;
++import org.bukkit.potion.PotionType;
++import org.jetbrains.annotations.ApiStatus;
++
++import java.util.List;
++import java.util.Set;
++
++@ApiStatus.Experimental
++public interface ItemProperties {
++
++    // COMMON
++    PropertyKey<List<EnchantmentPair>> ENCHANTMENTS = PropertyKey.of("Enchantments", TypeFactory.parameterizedClass(List.class, EnchantmentPair.class));
++    PropertyKey<Boolean> UNBREAKABLE = PropertyKey.of("Unbreakable", Boolean.class);
++    PropertyKey<List<Namespaced>> CAN_DESTROY_BLOCK_LIST = PropertyKey.of("CanDestroy", TypeFactory.parameterizedClass(List.class, Namespaced.class)); // TODO: BETTER REPRESENTATION
++    PropertyKey<List<Namespaced>> CAN_PLACE_ON_BLOCK_LIST = PropertyKey.of("CanPlaceOn", TypeFactory.parameterizedClass(List.class, Namespaced.class)); // TODO: BETTER REPRESENTATION
++    PropertyKey<Set<ItemFlag>> HIDE_FLAGS = PropertyKey.of("HideFlags", TypeFactory.parameterizedClass(Set.class, ItemFlag.class));
++    PropertyKey<Integer> REPAIR_COST = PropertyKey.of("RepairCost", Integer.class);
++    PropertyKey<Integer> CUSTOM_MODEL_DATA = PropertyKey.of("CustomModelData", Integer.class);
++    PropertyKey<Integer> DAMAGE = PropertyKey.of("Damage", Integer.class);
++    PropertyKey<List<AttributePair>> ATTRIBUTES = PropertyKey.of("AttributeModifiers", TypeFactory.parameterizedClass(List.class, AttributePair.class));
++    PropertyKey<MutablePropertyView> BLOCK_STATE = PropertyKey.of("BlockStateTag", MutablePropertyView.class);
++    PropertyKey<MutablePropertyView> ENTITY_TAG = PropertyKey.of("EntityTag", MutablePropertyView.class);
++
++    // DISPLAY PROPERTIES
++    PropertyKey<MutablePropertyView> DISPLAY = PropertyKey.of("display", MutablePropertyView.class);
```

Comment:
This should not have a property. Instead, the specific parts of display should have their own properties which point to `display.Name`, `display.Lore`, etc.